src/ray/common/ray_object.h

```diff
@@ -64,6 +64,10 @@ class RayObject {
   /// Whether the object represents an exception.
   bool IsException();
 
+  /// Whether the object has been promoted to plasma (i.e., since it was too
+  /// large to return directly as part of a gRPC response).
+  bool IsInPlasmaError();
```

Comment:
It is represented internally by an error, which is important to note.

Reply:
I don't think we should be exposing that implementation detail here

Reply:
You probably thought "this looks funny" when looking at this right? That is the intent. We are indeed doing something a bit weird here, which should be exposed as part of the API so that future developers get the right mental model of how direct call objects can indirectly refer to plasma objects.